libs/client/src/storage.ts

```diff
@@ -117,6 +117,9 @@ export const storageImpl: StorageSupport = {
         const url = await storageImpl.upload(blob as Blob);
         return [key, url];
       }
+      if (value !== null && typeof value === 'object') {
```

Comment:
`typeof value === 'object'` might return a few false positives, since instances of most classes are also objects.

We could introduce a utility in `utils.ts` that checks if it's a "plain object":

```ts
export function isPlainObject(value: any): boolean {
  return !!value && Object.getPrototypeOf(value) === Object.prototype;
}
```